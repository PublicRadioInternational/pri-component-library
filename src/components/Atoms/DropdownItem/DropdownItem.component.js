/**
 * @file DropdownItem.test.js
 * Exports a dropdown item component.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './DropdownItem.css';

/**
 * Component that renders a dropdown menu item.
 */
export default class DropdownItem extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    url: PropTypes.string,
    onClick: PropTypes.func
  };

  static defaultProps = {
    url: null,
    onClick: () => {}
  };

  render() {
    const { children, url, onClick } = this.props;
    return (
      <a className={styles.dropdownItem} href={url} onClick={onClick}>
        {children}
      </a>
    );
  }
}
