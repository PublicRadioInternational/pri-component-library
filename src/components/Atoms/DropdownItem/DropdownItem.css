/* import colors... */
@value colors: "../../00_global/colors.css";
@value blue, orange, white, grayLighter, blueLight, grayDark from colors;

/* import fonts... */
@value fonts: "../../00_global/fonts.css";
@value heading from fonts;

.dropdownItem,
.dropdownHeading,
.dropdownSpace {
  padding: 0.4rem 1.5rem;
}

.dropdownItem {
  clear: both;
  color: blue;
  display: block;
  font: 400 0.875em heading;
  width: 77%;
  border: none;
  cursor: pointer;
  text-decoration: none;
}

.dropdownItem:hover {
  background-color: blueLight;
  color: white;
}

.dropdownHeading {
  color: grayDark;
  font: 800 0.875em heading;
  font-weight: bold;
  margin: 0;
}

.dropdownHr {
  border: none;
  height: 1px;
  background-color: grayLighter;
}

.dropdownForm {
  width: 225px;
}
