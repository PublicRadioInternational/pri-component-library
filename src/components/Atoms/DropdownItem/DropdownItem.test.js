/**
 * @file DropdownItem.test.js
 * Contains tests for DropdownItem.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import DropdownItem from './DropdownItem.component';

describe('<DropdownItem />', () => {
  it('Handles Dropdown click events', () => {
    const onClick = jest.fn();
    const itemWrapper = shallow(
      <DropdownItem url="https://google.com" onClick={onClick}>
        Google
      </DropdownItem>
    );
    itemWrapper.find('a').simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(itemWrapper).toMatchSnapshot();
  });

  it('Matches the Dropdown Item Link snapshot', () => {
    const component = renderer.create(
      <DropdownItem url="https://google.com">Google</DropdownItem>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Matches the Dropdown Item Heading snapshot', () => {
    const component = renderer.create(<DropdownItem heading="Google" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Matches the Dropdown Item HR snapshot', () => {
    const component = renderer.create(<DropdownItem isHr />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Matches the Dropdown Item any children snapshot', () => {
    const component = renderer.create(<DropdownItem />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
