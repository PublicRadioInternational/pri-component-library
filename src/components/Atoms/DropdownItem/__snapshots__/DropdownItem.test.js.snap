// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem /> Matches the Link snapshot 1`] = `
<a
  className="dropdownItem"
  href="https://google.com"
  onClick={[MockFunction]}
>
  Google
</a>
`;
